nova_vm/src/ecmascript/builtins/builtin_function.rs

```diff
@@ -60,53 +81,196 @@ impl BuiltinFunctionArgs {
     }
 }
 
-/// ### [10.3.3 CreateBuiltinFunction ( behaviour, length, name, additionalInternalSlotsList \[ , realm \[ , prototype \[ , prefix \] \] \] )](https://tc39.es/ecma262/#sec-createbuiltinfunction)
+impl BuiltinFunctionIndex {
```

Comment:
To be frank, I'm not a fan of placing these as methods on the indexes. It seems to go against your original ideas of having standalone functions for these operations.

Reply:
You're absolutely correct. I should've instead created a struct that encapsulates the index and implemented the InternalMethods trait on it. I didn't because ECMAScriptFunction didn't and that was probably done just out of pure laziness.

I'll fix this to the proper form.